### ISSUE TITLE
Leak detect by strict mode

### ISSUE THREAD

**Issue by spvince** [GetStream/stream-chat-android#2697]
**Describe the bug**
Strict mode detect a leak

**SDK version**
- 4.22.0

**Leak**
```
D/StrictMode: StrictMode policy violation: android.os.strictmode.LeakedClosableViolation: A resource was acquired at attached stack trace but never released. See java.io.Closeable for information on avoiding resource leaks.
        at android.os.StrictMode$AndroidCloseGuardReporter.report(StrictMode.java:1924)
        at dalvik.system.CloseGuard.warnIfOpen(CloseGuard.java:303)
        at android.database.sqlite.SQLiteDatabase.dispose(SQLiteDatabase.java:344)
        at android.database.sqlite.SQLiteDatabase.finalize(SQLiteDatabase.java:328)
        at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:291)
        at java.lang.Daemons$FinalizerDaemon.runInternal(Daemons.java:278)
        at java.lang.Daemons$Daemon.run(Daemons.java:139)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.Throwable: Explicit termination method 'close' not called
        at dalvik.system.CloseGuard.openWithCallSite(CloseGuard.java:259)
        at dalvik.system.CloseGuard.open(CloseGuard.java:230)
        at android.database.sqlite.SQLiteDatabase.openInner(SQLiteDatabase.java:919)
        at android.database.sqlite.SQLiteDatabase.open(SQLiteDatabase.java:898)
        at android.database.sqlite.SQLiteDatabase.openDatabase(SQLiteDatabase.java:762)
        at android.database.sqlite.SQLiteDatabase.createInMemory(SQLiteDatabase.java:958)
        at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:368)
        at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:316)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableSupportDatabase(FrameworkSQLiteOpenHelper.java:145)
        at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.java:106)
        at io.getstream.chat.android.offline.repository.database.ChatDatabase_Impl.clearAllTables(ChatDatabase_Impl.java:393)
        at io.getstream.chat.android.offline.repository.builder.RepositoryFacadeBuilder$createDatabase$1$1.invokeSuspend(RepositoryFacadeBuilder.kt:44)
        at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
        at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
        at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
        at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
```

**Code**
```
        val client = clientBuilder
            .build()

        ChatDomain.Builder(client, context)
            .enableBackgroundSync()
            .recoveryEnabled()
            .offlineEnabled()
            .build()
```

**Expected behavior**
No leak

**Comment by ogkuzmin** [comment]
Thank you for submitting the issue! We will investigate it!

**Comment by adasiewiczr** [comment]
Hey @spvince ,
we have released a new SDK version (4.28.2) that contains a fix for this issue.
Feel free to reopen this ticket if the issue is not solved.
